arangod/Aql/MultiAqlItemBlockInputRange.cpp

```diff
@@ -35,12 +35,12 @@ using namespace arangodb::aql;
 
 namespace {
   static auto RowHasNonEmptyValue(ShadowAqlItemRow const& row) -> bool {
-    if (!row.isInitialized()) {
-      return false;
-    }
-    for (RegisterId registerId = 0; registerId < row.getNrRegisters(); ++registerId) {
-      if (!row.getValue(registerId).isEmpty()) {
-        return true;
+    if (row.isInitialized()) {
+      RegisterId const numRegisters = static_cast<RegisterId>(row.getNrRegisters());
```

Comment:
```suggestion
      auto const numRegisters = row.getNrRegisters();
```
should also work, if I'm not mistaken. But it's not important.